agent/api/statechange.go

```diff
@@ -101,12 +101,20 @@ type AttachmentStateChange struct {
 	Attachment *apieni.ENIAttachment
 }
 
+type ErrShouldNotSendEvent struct {
+	resourceId string
+}
+
+func (e ErrShouldNotSendEvent) Error() string {
+	return fmt.Sprintf("Should not send events for internal tasks or containers: %s", e.resourceId)
+}
+
 // NewTaskStateChangeEvent creates a new task state change event
 // returns error if the state change doesn't need to be sent to the ECS backend.
 func NewTaskStateChangeEvent(task *apitask.Task, reason string) (TaskStateChange, error) {
 	var event TaskStateChange
 	if task.IsInternal {
-		return event, errors.Errorf("skip creating task stage change event for internal task %v", task.Arn)
+		return event, ErrShouldNotSendEvent{task.Arn}
```

Comment:
Q. should we rename the error type or add more new error types since we don't send stake change under these 3 cases  
1. task.IsInternal 
2. !taskKnownStatus.BackendRecognized()
3.  task.GetSentStatus() >= taskKnownStatus

Reply:
Case #2 and #3 are a little different from case #1 imo. In case 1 we are dealing with special types of container/task, whereas in case 2 and 3 we are getting unexpected status which should not happen at all. And in such cases, we consider them real errors and log the message at error level (from the caller).

Reply:
Got it! Thanks for the clarification!